src/test/java/roomescape/infrastructure/JdbcReservationRepositoryTest.java

```diff
@@ -0,0 +1,121 @@
+package roomescape.infrastructure;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.ActiveProfiles;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationTime;
+import roomescape.domain.Theme;
+
+@JdbcTest
+@Import(JdbcReservationRepository.class)
+@ActiveProfiles("test")
+class JdbcReservationRepositoryTest {
+
+    @Autowired
+    private JdbcReservationRepository reservationRepository;
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @BeforeEach
+    void cleanDatabase() {
+        jdbcTemplate.execute("SET REFERENTIAL_INTEGRITY FALSE");
+        jdbcTemplate.execute("TRUNCATE TABLE reservation");
+        jdbcTemplate.execute("ALTER TABLE reservation ALTER COLUMN id RESTART WITH 1");
+        jdbcTemplate.execute("TRUNCATE TABLE reservation_time");
+        jdbcTemplate.execute("ALTER TABLE reservation_time ALTER COLUMN id RESTART WITH 1");
+        jdbcTemplate.execute("TRUNCATE TABLE theme");
+        jdbcTemplate.execute("ALTER TABLE theme ALTER COLUMN id RESTART WITH 1");
+        jdbcTemplate.execute("SET REFERENTIAL_INTEGRITY TRUE");
+    }
+
+    @DisplayName("save 후 생성된 id를 반환한다.")
+    @Test
+    void saveTest() {
+        // given
+        jdbcTemplate.update("INSERT INTO reservation_time (id, start_at) VALUES (1, '10:00')");
+        jdbcTemplate.update(
+                "INSERT INTO theme (id, name, description, thumbnail) VALUES (1, '테마1', '테마 1입니다.', '썸네일입니다.')");
+
+        Reservation reservation = Reservation.withoutId(
+                "테스트행님",
+                Theme.of(1L, "테마1", "테마 1입니다.", "썸네일"),
+                LocalDate.of(2025, 4, 30),
+                ReservationTime.of(1L, LocalTime.of(10, 0))
+        );
+
+        // when
+        Long savedId = reservationRepository.save(reservation);
+
+        // then
+        String sql = "SELECT * FROM reservation WHERE id = ?";
+        Map<String, Object> result = jdbcTemplate.queryForMap(sql, savedId);
+
+        assertAll(
+                () -> assertThat(savedId).isNotNull(),
+
+                () -> assertThat(result.get("name")).isEqualTo("테스트행님"),
```

Comment:
`containsEntry` 를 활용해볼 수 있습니다.
```suggestion
               () -> assertThat(result).containsEntry("name", "테스트행님"),
```

Reply:
`containsEntry()`가 있는지 몰랐는데 알려주셔서 감사합니다!
`containsEntry()`를 사용하면 (key, value) 쌍이 정확히 존재하는지 검증하는 과정에서 키의 존재 여부도 함께 확인하므로, `NullPointerException` 없이 테스트 실패로 안전하게 처리할 수 있다는 장점이 있다는 것을 알게 되었습니다.
또한 더 간결하고 명확하게 의도를 나타낼 수 있어 `containsEntry()`를 사용해 리팩터링했습니다 :)